### PR TITLE
=str Add java doc of Hub

### PR DESCRIPTION
**Pull request by He-Pin**
References #31137

### CODE REVIEW

**Review by johanandren — COMMENTED** [review]
Thanks, one tiny code-style issue.

File: akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala [hunk 1]
```diff
@@ -390,6 +390,11 @@ private[akka] class MergeHub[T](perProducerBufferSize: Int, drainingEnabled: Boo
  */
 object BroadcastHub {
 
+  /**
+   * INTERNAL API
+   */
+  @InternalApi private[akka] val defaultBufferSize = 256
```
[the remark this answers is not in the document]
Comment (jrudolph) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala:396]: Weird that we have this constant in code, but nothing to worry about in this PR.

**Review by jrudolph — APPROVED** [review]
LGTM

File: akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (johanandren) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala:396]: Actually found another buffer size in streams where we didn't uppercase like that, so let's leave it like this

**Review by johanandren — APPROVED** [review]

**Pull request merged**